poc_iot_injector/src/server.rs

```diff
@@ -123,10 +128,31 @@ async fn submit_txns(
             let mut shared_txn_service = txn_service.clone();
             let shared_key = keypair.clone();
             async move {
-                if let Ok(txn) =
-                    handle_txn_submission(msg, shared_key, &mut shared_txn_service, before_ts).await
-                {
-                    file_sink_write!("signed_poc_receipt_txn", receipt_sender, txn).await?;
+                match (
+                    handle_report_msg(msg, shared_key.clone(), before_ts),
+                    do_submission,
+                ) {
+                    (Ok(txn_details), true) => {
+                        // do txn submission
+                        if handle_txn_submission(txn_details.clone(), &mut shared_txn_service)
+                            .await
+                            .is_ok()
```

Comment:
handle_txn_submission will always return an ok response.  Should a failure scenario not be considered, for example what if the submit node is down or unreachable ?  Any txn submitted during this period will be lost ( in terms of not making it to the chain ).

Reply:
at the moment when we don't want the txns submitted to the chain for real logging the error seems like a decent enough approach but do we have a plan for handling failed txn construction or submission? either to store and retry later, crash, or something else?

Reply:
although `handle_txn_submission` can fail, so it's not guaranteed to return an `Ok`